src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -284,9 +292,22 @@ instance
     FromMnemonic (mw ': '[]) purpose
   where
     fromMnemonic parts = do
-        -- FIXME Do better than 'show' here
-        m <- first (FromMnemonicError . show) (mkMnemonic @mw parts)
+        m <- first (FromMnemonicError . pretty) (mkMnemonic @mw parts)
         return $ Passphrase $ entropyToBytes $ mnemonicToEntropy m
+      where
+        pretty = \case
+            ErrMnemonicWords ErrWrongNumberOfWords{} ->
+                "Invalid number of words: at least "
+                <> show (natVal (Proxy :: Proxy mw))
+                <> " words are expected."
+            ErrDictionary (ErrInvalidDictionaryWord w) ->
+                "Found invalid (non-English) word: \"" <> B.toList w <> "\""
+            ErrEntropy ErrInvalidEntropyChecksum{} ->
+                "Invalid entropy checksum: please double-check the last word of \
+                \your mnemonic sentence."
+            ErrEntropy ErrInvalidEntropyLength{} ->
+                "Something went wrong when trying to generate the entropy from \
+                \the given mnemonic. As a user, there's nothing you can do."
```

Comment:
Note that I am almost tempted to shove in a `error "OMG SOMETHING WENT WRONG"` here because really, we can never reach this branch. It'd mean we'd bypassed our `Mnemonic` type, which, as far as I am concerned, isn't possible currently.